pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/RRDD.scala

```diff
@@ -157,13 +158,21 @@ private abstract class BaseRRDD[T: ClassTag, U: ClassTag](
             dataOut.writeInt(1)
           }
 
-          for (elem <- iter) {
-            if (parentSerialized) {
-              val elemArr = elem.asInstanceOf[Array[Byte]]
-              dataOut.writeInt(elemArr.length)
-              dataOut.write(elemArr, 0, elemArr.length)
-            } else {
-              printOut.println(elem)
+          if (parentSerialized == "rows") {
```

Comment:
Thinking about this, it might be better to add an object in SerDe.scala which lists these serialization modes. something like 
```
object SerializationFormats {
   val ROWS = "rows"
   val BYTES = "bytes"
   val STRING = "string"
}
```